### PR TITLE
Health care questionnaire 24641 headers

### PR DESCRIPTION
**Pull request by mdewey**
## Description

Add the ability to customize the in-progress message for a form. Addresses some [accessiblity feedback](https://github.com/department-of-veterans-affairs/va.gov-team/issues/24641)

## Testing done (none yet) 
- [ ] unit 
- [ ] e2e

## Screenshots
screencapture-localhost-3001-health-care-health-questionnaires-questionnaires-answer-questions-introduction-2021-05-20-13_26_51

## Acceptance criteria
- [ ] "in-progress" message is an h2

## References
- [Ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/24641)
- [First PR](https://github.com/department-of-veterans-affairs/vets-website/pull/17279)

### CODE REVIEW

**Comment by mdewey** [comment]
@bkjohnson  This is in reference to [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/17279). This is using the parameter and ideas that came from Jesse's work. Was this what you had in mind? 

I was also thinking about using a different prop for the in-progress message, but that didn't feel right. What are your thoughts?

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx [hunk 1]
```diff
@@ -60,12 +60,15 @@ class SaveInProgressIntro extends React.Component {
 
         if (!isExpired) {
           const lastSavedDateTime = savedAt.format('M/D/YYYY [at] h:mm a');
-
+          const H = `h${this.props.headingLevel}`;
```
Comment (bkjohnson) [inline src/platform/forms/save-in-progress/SaveInProgressIntro.jsx:63]: `headingLevel` already exists as of #17282

(this is just a statement for my own sake, no response needed)

File: src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx [hunk 2]
```diff
@@ -82,6 +82,49 @@ describe('Schemaform <SaveInProgressIntro>', () => {
...
+        toggleLoginModal={toggleLoginModal}
+        formConfig={formConfig}
+        headingLevel={1}
+      />,
+    );
+
+    expect(tree.find('.usa-alert').text()).to.contain(
+      'Your application is in progress',
+    );
+    expect(tree.find('h1')).to.exist;
```
Comment (bkjohnson) [inline src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx:124]: Can this be taken a step further so that the expectation is that the `<h1>` node has `text()` which reads "Your application is in progress"?

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx [hunk 3]
```diff
@@ -60,12 +60,15 @@ class SaveInProgressIntro extends React.Component {
...
+          const message = `Your ${appType} is in progress`;
           alert = (
             <div>
               <div className="usa-alert usa-alert-info background-color-only schemaform-sip-alert">
                 <div className="schemaform-sip-alert-title">
-                  <strong>Your {appType} is in progress</strong>
+                  <H className="usa-alert-heading vads-u-font-size--base">
+                    <strong>{message}</strong>
```
Comment (bkjohnson) [inline src/platform/forms/save-in-progress/SaveInProgressIntro.jsx:70]: @joshkimux Do we still need a `<strong>` tag inside of whatever `<h*>` this ends up being?

**Review by bkjohnson — COMMENTED** [review]
Yeah, this is what I had in mind.

> I was also thinking about using a different prop for the in-progress message, but that didn't feel right. What are your thoughts?

I took a look locally at this file, and the extremely long conditional branching make things a bit hard to follow. This is a critique of the existing forms library, not this PR specifically. It _seems_ like the use of `headingLevel` in #17282 and the use of `headingLevel` that you're proposing with this PR are mutually exclusive - they both seem to be conditionally setting an `alert` variable returned by `getAlert()`, so I don't think there would be any interference. So, I don't think we need something like a `signedInHeadingLevel` and a `nonSignedInHeadingLevel` for each conditional branch, or something like that. Is this what you were asking?

File: src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx (hunk 2 above)
Comment (mdewey) [reply src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx]: Absolutely! https://github.com/department-of-veterans-affairs/vets-website/pull/17437/files#diff-a537e37b0547e6d4831e21f0318d284e75a9a91dc4b2e784f0135a5ae6619d64R121

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx (hunk 3 above)
Comment (mdewey) [reply src/platform/forms/save-in-progress/SaveInProgressIntro.jsx]: My logic in keeping the `<strong>` tag was to not break other tests. I just removed it and all the tests are passing. I do believe it's better to be removed.

File: src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx [hunk 4]
```diff
@@ -82,6 +82,49 @@ describe('Schemaform <SaveInProgressIntro>', () => {
...
+        toggleLoginModal={toggleLoginModal}
+        formConfig={formConfig}
+        headingLevel={1}
+      />,
+    );
+
+    expect(tree.find('h1.usa-alert-heading').text()).to.contain(
+      'Your application is in progress',
+    );
+    expect(tree.find('h1')).to.exist;
```
Comment (bkjohnson) [inline src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx:124]: I don't think we need the second `expect` now that the first one is specifically selecting a `h1`.

**Review by bkjohnson — APPROVED** [review]
I think we can simplify the test a bit, but everything else seems good :+1:

File: src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx (hunk 4 above)
Comment (mdewey) [reply src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx]: Removed :-)

**Review by dillo — APPROVED** [review]
